### PR TITLE
Enable Password Visibility During

### PR DESCRIPTION
**Pull request by Natali-Kotelnitska**
dev
Ticket  #609 

## Code reviewers

- [ ] @loqimean 
- [x] @DmytroStoliaruk 

## Summary of issue

Add a new feature in the login page that allows the user to toggle the visibility of their password.

## Summary of change
1. Add an eye icon in the password field in the pages with password fields.
2. Add styles for eye-icon.
3. Add enable password visibility logic using Stimulus.

Знімок екрана (251)
Знімок екрана (250)

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by DmytroStoliaruk — APPROVED** [review]

File: app/javascript/controllers/password_visibility_controller.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ['password', 'icon']
+
+  toggle() {
+    const icon = this.iconTarget
+    const passwordField = this.passwordTarget
```
Comment (loqimean) [inline app/javascript/controllers/password_visibility_controller.js:8]: а навіщо ця тавтологія?))

File: app/views/account/users/new.html.erb [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
       <%= f.input :email, placeholder: 'example@example.com' %>
       <%= f.input :first_name, placeholder: t('.form.first_name_label'), required: true %>
       <%= f.input :last_name, placeholder: t('.form.last_name_label'), required: true %>
       <%= f.input :country %>
       <%= f.input :role, collection: ['user', 'admin'] %>
-      <%= f.input :password, hint: t('.form.password_hint'), required: true %>
-      <%= f.input :password_confirmation, required: true %>
+
+      <div class="password-wrapper" data-controller="password-visibility">
+        <%= f.input :password, input_html: { data: { 'password-visibility-target': 'password' } }, hint: t('.form.password_hint'), required: true %>
```
Comment (loqimean) [inline app/views/account/users/new.html.erb:14]:
```suggestion
        <%= f.input :password, input_html: { data: { password_visibility_target: 'password' } }, hint: t('.form.password_hint'), required: true %>
```

File: app/views/account/users/new.html.erb [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
+        <%= render 'shared/password_icon' %>
+      </div>
+
+      <div class="password-wrapper" data-controller="password-visibility">
+        <%= f.input :password_confirmation, input_html: { data: { 'password-visibility-target': 'password' } }, required: true %>
```
Comment (loqimean) [inline app/views/account/users/new.html.erb:19]: тут те ж саме в `data`

Comment (loqimean) [reply app/views/account/users/new.html.erb:19]: і так скрізь

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/account/users/new.html.erb (hunk 2 above)
Comment (Natali-Kotelnitska) [reply app/views/account/users/new.html.erb:14]: без лапок вибиває синтаксичну помилку, а з лапками працює.

File: app/javascript/controllers/password_visibility_controller.js (hunk 1 above)
Comment (Natali-Kotelnitska) [reply app/javascript/controllers/password_visibility_controller.js]: Removed.

File: app/views/account/users/new.html.erb (hunk 3 above)
Comment (Natali-Kotelnitska) [reply app/views/account/users/new.html.erb]: Готово.

File: app/assets/stylesheets/components/form.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
+@layer components {
+
```
Comment (loqimean) [inline app/assets/stylesheets/components/form.css:2]:
```suggestion
```

File: app/javascript/controllers/password_visibility_controller.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ['password', 'icon']
+
+  toggle() {
+    const TEXT_INPUT = 'text';
+    const PASSWORD_INPUT = 'password';
```
Comment (loqimean) [inline app/javascript/controllers/password_visibility_controller.js:8]:
```suggestion
    const TEXT_INPUT_TYPE = 'text';
    const PASSWORD_INPUT_TYPE = 'password';
```

**Review by loqimean — APPROVED** [review]

**Pull request merged**